apps/angular/src/core/services/auth.service.ts

```diff
@@ -0,0 +1,90 @@
+import { catchError, map, Observable, switchMap, switchMapTo, throwError } from 'rxjs';
+
+import { Injectable } from '@angular/core';
+import { HttpClient } from '@angular/common/http';
+
+import { Login } from '@js-camp/core/models/login';
+import { TokenDto } from '@js-camp/core/dtos/token.dto';
+import { AppError } from '@js-camp/core/models/app-error';
+import { Registration } from '@js-camp/core/models/registration';
+import { TokensMapper } from '@js-camp/core/mappers/token.mapper';
+import { LoginDataMapper } from '@js-camp/core/mappers/login-data.mapper';
+import { RegistrationDataMapper } from '@js-camp/core/mappers/registration-data.mapper';
+
+import { UrlService } from './url.service';
+import { TokenService } from './token.service';
+import { AppConfigService } from './app-config.service';
+
+const UNAUTHORIZED_ERROR = 'User unauthorized';
```

Comment:
```suggestion
const UNAUTHORIZED_ERROR_MESSAGE = 'User unauthorized';
```
It's just a string, not an `Error` object :)